File: src/api/v2/analytics.ts
import v2Client from './client';
import type { DashboardMetrics, VelocityMetric, ReleaseTrendMetric, EndorsementInsight, RiskItem } from './types';

export const analyticsAPI = {
    getDashboardMetrics: async (): Promise<DashboardMetrics> => {
        const response = await v2Client.get('/analytics/dashboard');
        return response.data as DashboardMetrics;
    },

    async getReleaseTrends(): Promise<ReleaseTrendMetric[]> {
        const response = await v2Client.get('/analytics/trends/releases');
        return response.data as ReleaseTrendMetric[];
    },
    async getEndorsementInsights(): Promise<EndorsementInsight[]> {
        const response = await v2Client.get('/analytics/insights/endorsements');
        return response.data as EndorsementInsight[];
    },
    async getRiskHeatmap(): Promise<RiskItem[]> {
        const response = await v2Client.get('/analytics/heatmap/risk');
        return response.data as RiskItem[];
    },

    getVelocityMetrics: async (): Promise<VelocityMetric[]> => {
        const response = await v2Client.get('/analytics/velocity');
        return response.data as VelocityMetric[];
    },
};

File: backend/app/api/v2/endpoints/analytics.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.api import deps
from app.services import analytics_service

router = APIRouter()

@router.get("/dashboard")
async def get_dashboard_metrics(
    db: Session = Depends(deps.get_db)
):
    return await analytics_service.get_dashboard_metrics(db)

@router.get("/velocity")
async def get_velocity_metrics(
    db: Session = Depends(deps.get_db)
):
    return await analytics_service.get_velocity_metrics(db)

@router.get("/trends/releases")
async def get_release_trends(
    db: Session = Depends(deps.get_db)
):
    return await analytics_service.get_release_trends(db)

@router.get("/insights/endorsements")
async def get_endorsement_insights(
    db: Session = Depends(deps.get_db)
):
    return await analytics_service.get_endorsement_insights(db)

@router.get("/heatmap/risk")
async def get_risk_heatmap(
    db: Session = Depends(deps.get_db)
):
    return await analytics_service.get_risk_heatmap(db)
